### PR TITLE
For #216: Decklist displays current hero deck

### PR DESCRIPTION
**Pull request by paulodamaso**
For #216: Decklist displays current hero deck

- Added DeckListViewCards script which will get all hero cards from party and put them into DeckList
- Added List<Card > to Hero so we can get that hero specific set of cards
- Just couldn't wire it to appear in screen

### CODE REVIEW

File: src/ProjectNeon/Assets/Data/Templates/Character.cs [hunk 1]
```diff
@@ -1,8 +1,14 @@
...
+using UnityEngine;
 
 public class Character : ScriptableObject
 {
     public Sprite Bust;
     public Stats Stats;
     public string ClassName;
+
+    [SerializeField]
+    private List<Card> cards;
```
Comment (SilasReinagel) [inline src/ProjectNeon/Assets/Data/Templates/Character.cs]: Characters are supposed to be Design-Time unchangeable reference data.

Decks are mutable and will change during the game. `PartyDecks` is the state structure you are looking for.

File: src/ProjectNeon/Assets/DeckListViewCards.cs [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+using UnityEngine;
+
+public class DeckListViewCards : MonoBehaviour
+{
+    [SerializeField]
+    private Party party;
+
+    private List<Card> cards;
+
+    public void Init()
```
Comment (SilasReinagel) [inline src/ProjectNeon/Assets/DeckListViewCards.cs]: When is this called?

File: src/ProjectNeon/Assets/PartyDeckList.cs [hunk 3]
```diff
@@ -0,0 +1,8 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class PartyDeckList : MonoBehaviour
```
Comment (SilasReinagel) [inline src/ProjectNeon/Assets/PartyDeckList.cs]: What purpose does this class serve?

**Comment by paulodamaso** [comment]
@SilasReinagel Please take a look to my new implementation:
- I created `DeckController` which will control Deck  operations inside `DeckBuilderScene`. `DeckController ` will have a pointer to current Hero deck, extracted from `PartyDecks`
- Created `ChangeCurrentCharacter` event; this event will be fired on character change and will trigger  `DeckController` method `OnHeroChange` which will display the hero cards on the card list

What I couldn't achieve and need help:
- Wire `ChangeCurrentCharacter` to `DeckController.OnHeroChange` execution
- Add `Card` objects dinamically to `DeckListView`. My doubt here is that these objects are unity objects and I don't know how to instantiate them properly dynamically in my c# scripts

File: src/ProjectNeon/Assets/Features/Battle/Events/ChangeCurrentCharacter.asset [hunk 4]
```diff
@@ -0,0 +1,14 @@
+%YAML 1.1
```
Comment (SilasReinagel) [inline src/ProjectNeon/Assets/Features/Battle/Events/ChangeCurrentCharacter.asset]: This is not a battle event. This is a DeckBuilder event.

File: src/ProjectNeon/Assets/Features/DeckBuilder/Scripts/DeckController.cs [hunk 5]
```diff
@@ -0,0 +1,23 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using static CharacterController;
+
+public class DeckController : MonoBehaviour
```
Comment (SilasReinagel) [inline src/ProjectNeon/Assets/Features/DeckBuilder/Scripts/DeckController.cs]: Should avoid using words like `Manager` and `Controller`. Should be scoped to fixed and clear responsibilities.

File: src/ProjectNeon/Assets/Features/DeckBuilder/Scripts/DeckController.cs [hunk 6]
```diff
@@ -0,0 +1,23 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using static CharacterController;
```
Comment (SilasReinagel) [inline src/ProjectNeon/Assets/Features/DeckBuilder/Scripts/DeckController.cs]: Never use static imports.

File: src/ProjectNeon/Assets/Features/DeckBuilder/Scripts/DeckController.cs [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+{
+    [SerializeField] private PartyDecks decks;
+
+    private Deck current;
+
+    public void OnHeroChange()
+    {
+        this.current = decks.Decks[
+            Array.IndexOf(
+                Enum.GetValues(CharacterController.currentCharacter.GetType()),
```
Comment (SilasReinagel) [inline src/ProjectNeon/Assets/Features/DeckBuilder/Scripts/DeckController.cs]: We do not want to depend on a non-state type.

DeckBuilder State should live in a DeckBuilderState scriptable object. 

If there is to be a current character, it should be there.

File: src/ProjectNeon/Assets/Features/DeckBuilder/Scripts/DeckController.cs [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+public class DeckController : MonoBehaviour
+{
+    [SerializeField] private PartyDecks decks;
+
+    private Deck current;
+
+    public void OnHeroChange()
+    {
+        this.current = decks.Decks[
+            Array.IndexOf(
```
Comment (SilasReinagel) [inline src/ProjectNeon/Assets/Features/DeckBuilder/Scripts/DeckController.cs]: This Character enum needs to die. It is unscalable.

**Comment by paulodamaso** [comment]
@SilasReinagel Made another PR, #259

**Pull request closed without merging**